core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp

```diff
@@ -483,11 +483,19 @@ struct reduction_identity<Kokkos::Experimental::bhalf_t> {
   KOKKOS_FORCEINLINE_FUNCTION constexpr static float prod() noexcept {
     return 1.0F;
   }
-  KOKKOS_FORCEINLINE_FUNCTION constexpr static float max() noexcept {
-    return -Kokkos::Experimental::infinity_v<float>;
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static auto max() noexcept {
+#if !__FINITE_MATH_ONLY__
+    return -Kokkos::Experimental::infinity_v<Kokkos::Experimental::bhalf_t>;
+#else
+    return Kokkos::Experimental::finite_min_v<Kokkos::Experimental::bhalf_t>;
+#endif
```

Comment:
Readability:
1. avoid if not A, else B.  Do if B else A
2. (optional) w/o the Kokkos::Experimental:: qualification it is slightly more readable, the minus sign is closer to inf and there is less boiler plate

```suggestion
    using namespace Kokkos::Experimental;
#if __FINITE_MATH_ONLY__
    return finite_min_v<bhalf_t>;
#else
    return -infinity_v<bhalf_t>;
#endif
```